### ISSUE TITLE
Improve file search by ignoring diacritical marks

### ISSUE THREAD

**Issue by guiweber** [RetroShare/RetroShare#118]
When searching for files under File Tranfers, characters with diacritical marks such as accents, cedillas, umlauts and tildes (e.g. é, è, ê, ü, ç, ñ...) are treated as discrete characters. Treating these accented characters the same as the base characters would make searching easier.

For example, if I share a video of Björk with the accented "ö" in the file name and someone searches for "bjork", he will get no results. It would be better if searching for "o" returned results for that letter with any possible diacritical mark. An option to only match the exact characters could be added to the advanced search options if necessary, in a way similar to the "match case" option.

There are several scenarios where people are unlikely to know the right diacritical marks to use (foreign artists for example). There are also situations when searching with the right diacritical marks may return no results, as some people omit them on purpose from their file names for compatibility or other reasons.

Thanks!

**Comment by csoler** [comment]
That has been suggested already. It needs a proper comparison function between strings in the code that select files from the shared lists. But it's going to raise the cost of handling file searches.

**Comment by TheSentry** [comment]
Alternatively, each file (or generally: item) with special characters gets two entries in the index, one unaltered and one with all special characters converted to their base character.

All we need for transfers is the hash, so after the apropriate hash has been located in the index, we can proceed as usual.

**Comment by csoler** [comment]
makes sense. I'll keep that in mind when re-coding the file list system (soon).
